addons/mail/static/src/models/thread/thread.js

```diff
@@ -1615,6 +1619,10 @@ function factory(dependencies) {
                 'serverFoldState',
             ],
         }),
+        /**
+         * All messages ordered like they are displayed. This field does not
+         * contain transient messages which are not "real" records.
+         */
```

Comment:
Why ignore transient messages here? This doesn't look right at a glance.

Reply:
I guess it has to do with `orderedMessages` being used for unread and seen, and code breaks with transient messages.
Code that needs ordered messages without transient should have a field named appropriately.
I wouldn't mind having 2 fields `orderedMessages` and `orderedNonTransientMessages`.

Reply:
We had discussion before. We don't ever need the field with with the transient messages.

The only place where we use transient them is from thread cache to display them on message list.

Reply:
`orderedMessages` should just be `messages` with some order, the fact it filters some messages mean that it should have a more specific name.

Reply:
so ?

Reply:
Could rename this field to `orderedNonTransientMessages`.
I definitely don't like `orderedMessages` with no transient messages.